sdks/python/apache_beam/io/kinesis.py

```diff
@@ -0,0 +1,317 @@
+  * Update your pipeline to provide the expansion service address when
+    initiating Kinesis transforms provided in this module.
+
+  Flink Users can use the built-in Expansion Service of the Flink Runner's
+  Job Server. If you start Flink's Job Server, the expansion service will be
+  started on port 8097. For a different address, please set the
+  expansion_service parameter.
+
+  **More information**
+
+  For more information regarding cross-language transforms see:
+  - https://beam.apache.org/roadmap/portability/
+
+  For more information specific to Flink runner see:
+  - https://beam.apache.org/documentation/runners/flink/
+"""
+
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import time
+from typing import List
+from typing import NamedTuple
+from typing import Optional
+from typing import Tuple
+
+from past.builtins import unicode
+
+from apache_beam import BeamJarExpansionService
+from apache_beam import ExternalTransform
+from apache_beam import NamedTupleBasedPayloadBuilder
+
+__all__ = [
+    'WriteToKinesis',
+    'ReadDataFromKinesis',
+    'InitialPositionInStream',
+    'WatermarkPolicy',
+]
+
+
+def default_io_expansion_service():
+  return BeamJarExpansionService(
+      'sdks:java:io:kinesis:expansion-service:shadowJar')
+
+
+WriteToKinesisSchema = NamedTuple(
+    'WriteToKinesisSchema',
+    [
+        ('stream_name', unicode),
+        ('aws_access_key', unicode),
+        ('aws_secret_key', unicode),
+        ('region', unicode),
+        ('partition_key', unicode),
+        ('service_endpoint', Optional[unicode]),
+        ('verify_certificate', Optional[bool]),
+        ('producer_properties', Optional[List[Tuple[unicode, unicode]]]),
+    ],
+)
+
+
+class WriteToKinesis(ExternalTransform):
+  """
+    An external PTransform which writes byte array stream to Amazon Kinesis.
+
+    Experimental; no backwards compatibility guarantees.
+  """
+  URN = 'beam:external:java:kinesis:write:v1'
+
+  def __init__(
+      self,
+      stream_name,
+      aws_access_key,
+      aws_secret_key,
+      region,
+      partition_key,
+      service_endpoint=None,
+      verify_certificate=None,
+      producer_properties=None,
+      expansion_service=None,
+  ):
+    """
+    Initializes a write operation to Kinesis.
+
+    :param stream_name: Kinesis stream name.
+    :param aws_access_key: Kinesis access key.
+    :param aws_secret_key: Kinesis access key secret.
+    :param region: AWS region. Example: 'us-east-1'.
+    :param service_endpoint: Kinesis service endpoint
+    :param verify_certificate: Enable or disable certificate verification.
+        Never set to False on production. True by default.
+    :param partition_key: Specify default partition key.
+    :param producer_properties: Specify the configuration properties for Kinesis
+        Producer Library (KPL) as List[KV[string, string]].
+        Example: [('CollectionMaxCount', '1000'), ('ConnectTimeout', '10000')]
```

Comment:
Can you make this argument accept a mapping and convert it to `List[Tuple[..]]` as is done in KafkaIO?

Reply:
Done